### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by e1528532**
# Purpose

Started improving some general documentation things i encountered when first trying elektra. Still WIP.

# Checklist

- [X] commit messages are fine (with references to issues)
- [X] affected documentation is fixed
- [X] meta data is updated (e.g. README.md of plugins)

@markus2330 since you encouraged me also to make pull requests for WIP things, here it is ;). I yet have to go over the usage of key-path / key-name etc. once again, with my better understanding of the difference now. Since i struggled a lot more with getting the java parts to run, improvements on the java documentation will follow in a separate branch/pull requests.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -212,7 +212,7 @@ For [OpenSUSE, CentOS, Fedora, RHEL and SLE](https://build.opensuse.org/package/
 Kai-Uwe Behrmann kindly provides packages [for download](http://software.opensuse.org/download.html?project=home%3Abekun%3Adevel&package=libelektra4).
 For Debian wheezy and jessie amd64 we provide latest builds. See build server below.
 
-If there are no packages available for your distribution, see the
+If there are no packages available for your distribution, or if you are using a different operating system like macOS, see the
```
Comment (markus2330) [inline README.md]: macOS now has homebrew packages, I think we can recommend that now.

Please also try it and give feedback.

File: doc/help/kdb-editor.md [hunk 2]
```diff
@@ -89,3 +89,4 @@ Or set a new editor as default using:
 ## SEE ALSO
 
 - [elektra-merge-strategy(7)](elektra-merge-strategy.md)
+- [elektra-key-names(7)](elektra-key-names.md) for an explanation of key names.
```
Comment (markus2330) [inline doc/help/kdb-editor.md]: Please use key-?name consistently, discussion about it in #920

File: src/tools/kdb/cmdline.cpp [hunk 3]
```diff
@@ -517,7 +517,11 @@ kdb::Key Cmdline::createKey (int pos) const
 
 	if (!root.isValid ())
 	{
-		throw invalid_argument (name + " is not a valid keyname");
+		throw invalid_argument (name + " is not a valid keyname" +
+								"\n\n"
+								"Please make sure that the key's namespace is valid and not missing (see 'man elektra-namespaces').\n"
```
Comment (markus2330) [inline src/tools/kdb/cmdline.cpp]: "not missing" is misleading, in cascading keys it is kind of "missing"?

File: README.md (hunk 1 above)
Comment (e1528532) [inline README.md]: ah yeah, i added it to the package list. 
Encountered a small issue with the brew package https://github.com/ElektraInitiative/libelektra/issues/1243 which shouldnt be a problem though for normal users.

I was not sure how to control which optional components of elektra shall be built using the homebrew package but after a small study on the homebrew issue https://github.com/ElektraInitiative/libelektra/issues/531 i found out that i can see the parameters via brew info elektra. Is it worth mentioning this in the readme of our tap?

File: README.md [hunk 4]
```diff
@@ -212,7 +212,10 @@ For [OpenSUSE, CentOS, Fedora, RHEL and SLE](https://build.opensuse.org/package/
 Kai-Uwe Behrmann kindly provides packages [for download](http://software.opensuse.org/download.html?project=home%3Abekun%3Adevel&package=libelektra4).
 For Debian wheezy and jessie amd64 we provide latest builds. See build server below.
 
-If there are no packages available for your distribution, or if you are using a different operating system like macOS, see the
+For macOS there is a homebrew tap available:
+  - [macOS](https://github.com/ElektraInitiative/homebrew-elektra)
+
+If there are no packages available for your distribution see the
```
Comment (markus2330) [inline README.md]: Shouldn't we link to doc/COMPILE.md?

File: src/tools/kdb/cmdline.cpp (hunk 3 above)
Comment (e1528532) [inline src/tools/kdb/cmdline.cpp]: true. i adjusted the wording a bit, the problem is basically when you try to use a non-cascaded key, as i initally kept wondering why creating a key test/key did not work (> as there is most likely no test namespace globally). user/test would have worked instead, as the user namespaces does exist. 

is it better to understand now? any special situations i'm not yet aware about?

File: src/tools/kdb/cmdline.cpp [hunk 5]
```diff
@@ -518,8 +518,8 @@ kdb::Key Cmdline::createKey (int pos) const
 	if (!root.isValid ())
 	{
 		throw invalid_argument (name + " is not a valid keyname" +
-								"\n\n"
-								"Please make sure that the key's namespace is valid and not missing (see 'man elektra-namespaces').\n"
+								"\n\n" +
+								"For absolute keys (starting without /), please note that only one of the predefined namespaces can be used (see 'man elektra-namespaces').\n" +
```
Comment (markus2330) [inline src/tools/kdb/cmdline.cpp]: What about giving an example of an valid keyname?

Comment (e1528532) [inline src/tools/kdb/cmdline.cpp]: i added both examples for a cascading and a non-cascading key

**Comment by e1528532** [comment]
So I've revised the naming again. key name should be used consistently now, and keeping the explanation of key name vs path again and hopefully used it properly now. As a key name sort of represents a path its not always the easiest thing to decide what to use, but i try to use key name only when it actually points to a specific key and not to paths/hierarchies (so multiple keys might be affected by the operation or no actual key gets created like for mountpoints).

So if its ok now, i think this is pretty much ready for merge.

**Comment by markus2330** [comment]
Thank you! Please avoid so many merges in future and rebase instead.

**Pull request merged**